module/spl/spl-taskq.c

```diff
@@ -684,6 +685,18 @@ taskq_dispatch_ent(taskq_t *tq, task_func_t func, void *arg, uint_t flags,
 	spin_lock(&t->tqent_lock);
 
 	/*
+	 * Make sure the entry is not on some other taskq; it is important to
+	 * do this under lock
+	 */
+	ASSERT(taskq_empty_ent(t));
+	if (!taskq_empty_ent(t)) {
+		cmn_err(CE_WARN, "taskq_dispatch_ent(): entry %p is not empty -"
+			" prev %p, next %p\n",
+			&t->tqent_list, t->tqent_list.prev, t->tqent_list.next);
+		spl_dumpstack();
+	}
```

Comment:
I suppose it would.  But is this something we suspect is happening in productions very rarely but not during any of the automated testing?